linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -767,38 +767,25 @@ private final class Analyzer(config: CommonPhaseConfig,
 
       val candidates = ancestorsInReflectiveTargetOrder.iterator.map(_.findProxyMatch(proxyName))
 
-      val targetFuture = locally {
-        implicit val iec = ec
-
-        /* Manual version of
-         * Future.sequence(candidates).map(_.collectFirst { case Some(m) => m })
-         *
-         * We use a manual version because `Future.sequence` uses zipWith, not
-         * flatMap, and therefore does not wait for one future to complete
-         * before continuing on to the next one. This is generally good
-         * behavior, but it kills our use case, as it bypasses the
-         * short-circuiting behavior of `Iterator`s, resulting in *a lot* of
-         * useless computations!
-         */
-        def loop(): Future[Option[MethodInfo]] = {
+      val optTargetFuture = {
+        @tailrec
+        def loop(): Option[Future[MethodInfo]] = {
```

Comment:
At this point, you don't need the handrolled loop anymore, right? (this is just a "get the first Some in an iterator").

Does it make sense to drop the first commit of #4811?